internal/neofs/neofs.go

```diff
@@ -316,8 +325,12 @@ func (x *NeoFS) CreateObject(ctx context.Context, prm layer.PrmObjectCreate) (oi
 		return oid.ID{}, fmt.Errorf("save object via connection pool: %w", err)
 	}
 
-	chunk := make([]byte, x.cfg.MaxObjectSize)
-	_, err = io.CopyBuffer(writer, prm.Payload, chunk)
+	data := x.buffers.Get()
+	chunk := data.(*[]byte)
```

Comment:
Likely some Go 1.2X will bring generic-based `Pool` to us.

Reply:
It would be great, but 1.21 still have `any` version